source/Command.cpp

```diff
@@ -355,15 +355,21 @@ Command::operator bool() const
 // Check whether this command is entirely empty.
 bool Command::operator!() const
 {
-	return !state && !turn;
+	return state.none() && !turn;
 }
 
 
 
 // For sorting commands (e.g. so a command can be the key in a map):
 bool Command::operator<(const Command &command) const
 {
-	return (state < command.state);
+	static const int count = static_cast<int>(COMMAND_COUNT);
```

Comment:
I don't really see the point for this. You can use `.size()` here too and make `i` a size_t as well.

Reply:
The `i` can't be an unsigned type because the `for(...;--i)` would loop forever. For the loop to stop, `i` must reach -1 at the end of a loop iteration.

Reply:
oh whoops didn't even notice. That's annoying. I guess you could change it in such a way that you use `i - 1` instead in body and change the loop appropriately. You decide though

Reply:
I could also change it to be `i < command.state.size()` and really confuse people. However, I think that limiting us to `2**31-1` commands is an acceptable trade for understandability.